nsxt/policy_common.go

```diff
@@ -282,19 +288,28 @@ func getSecurityPolicyAndGatewayRulesSchema(scopeRequired bool, isIds bool, nsxI
 	if isIds {
 		ruleSchema["ids_profiles"] = getIdsProfilesSchema()
 	}
-	return &schema.Schema{
-		Type:        schema.TypeList,
-		Description: "List of rules in the section",
-		Optional:    true,
-		MaxItems:    1000,
-		Elem: &schema.Resource{
-			Schema: ruleSchema,
-		},
+	if separated {
+		ruleSchema["policy_path"] = getPolicyPathSchema(true, false, "Security Policy path")
+		ruleSchema["sequence_number"] = &schema.Schema{
+			Type:        schema.TypeInt,
+			Description: "Sequence number of the this rule",
+			Required:    true,
+		}
+		ruleSchema["context"] = getContextSchema()
```

Comment:
It makes sense to specify the context for sake of consistency, but its also true that context can be derived from the policy object. Not sure what is the best approach here - specifying the context for hundreds of rules can be an overhead for the user

Reply:
Changed. If users didn't provide the context, then we will extract the `project_id` from `policy_path` and fill the `context`.